src/main/kotlin/auth/BasicAuth.kt

```diff
@@ -0,0 +1,78 @@
+package se.zensum.leia.auth
+
+import com.mantono.pyttipanna.HashAlgorithm
+import com.mantono.pyttipanna.hash
+import io.ktor.util.decodeBase64
+import leia.logic.IncomingRequest
+import mu.KotlinLogging
+import org.apache.commons.codec.binary.Hex
+import java.util.*
+
+private val log = KotlinLogging.logger("basic-auth")
+
+private const val HEADER = "Authorization"
+
+class BasicAuth(credentials: Map<String, String>): AuthProvider {
+
+    private val credentials: Map<String, ByteArray> = credentials.mapValues {
+        Hex.decodeHex(it.value)
+    }
+
+    init {
+        this.credentials.values.forEach { hash ->
+            require(hash.size == 32) {
+                "Input for hashed passwords is expected to be SHA256 of 32 bytes length (was ${hash.size})"
+            }
+        }
+    }
+
+
+
+    override fun verify(matching: List<String>, incomingRequest: IncomingRequest) : AuthResult {
+        if(!incomingRequest.headers.containsKey(HEADER))
+            return AuthResult.Denied
+
+        val credential: String = incomingRequest.headers[HEADER]!!
```

Comment:
I should write a test case for this scenario. I think it will include `Bearer ` in the string that it will try to decode as base64, which will throw an exception. I do not think that is the behavior we want ;) I'll change it to deny if missing "Basic".

Reply:
Never mind, it worked as it should, I had already thought of this. Good to have a test for it tough.